migrations/migration_encrypt_shares.go

```diff
@@ -0,0 +1,47 @@
+package migrations
+
+import (
+	"context"
+	"crypto/sha256"
+	"crypto/x509"
+	"fmt"
+
+	"go.uber.org/zap"
+)
+
+var encryptSharesMigration = Migration{
+	Name: "encrypt_shares",
+	Run: func(ctx context.Context, logger *zap.Logger, opt Options, key []byte) error {
+		nodeStorage, err := opt.nodeStorage(logger)
+		if err != nil {
+			return fmt.Errorf("failed to get node storage: %w", err)
+		}
+		signerStorage := opt.signerStorage(logger)
+		accounts, err := signerStorage.ListAccounts()
+		if err != nil {
+			return fmt.Errorf("failed to list accounts: %w", err)
+		}
+
+		operatorKey, found, err := nodeStorage.GetPrivateKey()
+		if err != nil {
+			return fmt.Errorf("failed to get private key: %w", err)
+		}
+		if !found {
+			return nil
+		}
```

Comment:
just for the sake of order, I would do it like this 
```suggestion
		nodeStorage, err := opt.nodeStorage(logger)
		if err != nil {
			return fmt.Errorf("failed to get node storage: %w", err)
		}
		operatorKey, found, err := nodeStorage.GetPrivateKey()
		if err != nil {
			return fmt.Errorf("failed to get private key: %w", err)
		}
		if !found {
			return nil
		}
		signerStorage := opt.signerStorage(logger)
		accounts, err := signerStorage.ListAccounts()
		if err != nil {
			return fmt.Errorf("failed to list accounts: %w", err)
		}
```